js/src/Graph.js

```diff
@@ -366,36 +366,17 @@ var Graph = mark.Mark.extend({
     },
 
     process_interactions: function() {
-        var interactions = this.model.get("interactions");
-        if(_.isEmpty(interactions)) {
-            //set all the event listeners to blank functions
-            this.reset_interactions();
-        } else {
-            if(interactions.click !== undefined &&
-               interactions.click !== null) {
-                if(interactions.click === "tooltip") {
-                    this.event_listeners.element_clicked = function() {
-                        return this.refresh_tooltip(true);
-                    };
-                    this.event_listeners.parent_clicked = this.hide_tooltip;
-                } else if (interactions.click == "select") {
-                    this.event_listeners.parent_clicked = this.reset_selection;
-                    this.event_listeners.element_clicked = this.click_handler;
-                }
-            } else {
-                this.reset_click();
-            }
-            if(interactions.hover !== undefined &&
-              interactions.hover !== null) {
-                if(interactions.hover === "tooltip") {
-                    this.event_listeners.mouse_over = this.refresh_tooltip;
-                    this.event_listeners.mouse_move = this.move_tooltip;
-                    this.event_listeners.mouse_out = this.hide_tooltip;
-                }
-            } else {
-                this.reset_hover();
+        Graph.__super__.process_interactions.apply(this);
+        const interactions = this.model.get("interactions");
+
+        if(interactions.click !== undefined &&
+           interactions.click !== null) {
+            if (interactions.click == "select") {
+                this.event_listeners.parent_clicked = this.reset_selection;
+                this.event_listeners.element_clicked = this.click_handler;
```

Comment:
All of the marks which overload this interaction only use it to set a custom `click_handler`. We could generalise this just as we do for `reset_selection` to remove the need to overload at all in many cases.